lightning/mydump/parser.go

```diff
@@ -160,6 +193,57 @@ func (parser *blockParser) readBlock() error {
 	}
 }
 
+var unescapeRegexp = regexp.MustCompile(`(?s)\\.`)
+
+func unescape(
+	input string,
+	delim string,
+	escFlavor backslashEscapeFlavor,
+) string {
+	if len(delim) > 0 {
+		delim2 := delim + delim
+		if strings.Index(input, delim2) != -1 {
+			input = strings.Replace(input, delim2, delim, -1)
+		}
+	}
+	if escFlavor == backslashEscapeFlavorMySQLWithNull && input == `\N` {
+		return legacyNullSequence
+	}
+	if escFlavor != backslashEscapeFlavorNone && strings.IndexByte(input, '\\') != -1 {
+		input = unescapeRegexp.ReplaceAllStringFunc(input, func(substr string) string {
+			switch substr[1] {
+			case '0':
+				return "\x00"
+			case 'b':
+				return "\b"
+			case 'n':
+				return "\n"
+			case 'r':
+				return "\r"
+			case 't':
+				return "\t"
+			case 'Z':
+				return "\x26"
+			default:
+				return substr[1:]
```

Comment:
is it ok to discard '\\'？

Reply:
`\\` is would be unescaped as `\` just like all other normal characters.

```
mysql> select '\\\x\y\1';
+------+
| \xy1 |
+------+
| \xy1 |
+------+
1 row in set (0.00 sec)
```

Reply:
ok